File: src/main/java/com/heybys/oddments/fooddelivery/domain/generic/Address.java
package com.heybys.oddments.fooddelivery.domain.generic;

import com.heybys.oddments.base.domain.ValueObject;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.Getter;

@Getter
@Embeddable
public class Address extends ValueObject<Address> {

    private String city;

    private String street;

    private String zipCode;

    public Address() {}

    public Address(String city, String street, String zipCode) {
        this.city = city;
        this.street = street;
        this.zipCode = zipCode;
    }

    @Override
    public boolean equals(Object other) {
        return super.equals(other);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/domain/generic/Address.kt
package com.heybys.oddments.fooddelivery.domain.generic

import com.heybys.oddments.base.domain.ValueObject
import jakarta.persistence.Embeddable

@Embeddable
class Address(val city: String? = null, val street: String? = null, val zipCode: String? = null) :
    ValueObject<Address>() {

    // JPA constructor
    constructor() : this(null, null, null)

    override fun equals(other: Any?): Boolean = super.equals(other)

    override fun hashCode(): Int = super.hashCode()
}
